lib/internal/modules/esm/fetch_module.js

```diff
@@ -107,7 +107,11 @@ function fetchWithRedirects(parsed) {
     // `finally` on network error/timeout.
     const { 0: res } = await once(req, 'response');
     try {
-      const isRedirect = res.statusCode >= 300 && res.statusCode <= 303;
+      const isRedirect = res.statusCode >= 300 && (
+        res.statusCode <= 303 ||
+        res.statusCode === 307 ||
+        res.statusCode === 308
+      );
```

Comment:
It’s perhaps time to split `isRedirect` into its own function, with a JSDoc comment explaining the various error codes, with links.

Reply:
I'm not particularly familiar with the codebase, should I put the function in this file or some other file?

Reply:
This file is fine, since it’s not used elsewhere. Thanks!